docs/data-sources/user.md

```diff
@@ -0,0 +1,47 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
```

Comment:
Since the official doc generator doesn't support setting `subcategory`, the docs here require a custom file in https://github.com/goauthentik/terraform-provider-authentik/tree/main/templates/data-sources
(I need to update the datasource templates, but use something like https://github.com/goauthentik/terraform-provider-authentik/blob/main/templates/resources/application.md.tmpl as reference)

Reply:
Hi @BeryJu, thank you for the review, I pushed a few modifications, including the custom documentation files.

I think I also found a way to make the failing test pass.

Another MR will be opened shortly for an update of the `authentik_group` data source and a new `authentik_groups` data source.